itest/rust/src/object_tests/property_test.rs

```diff
@@ -351,20 +354,101 @@ fn derive_export() {
         .iter_shared()
         .find(|c| c.get_or_nil("name") == "foo".to_variant())
         .unwrap();
-    assert_eq!(
-        property.get_or_nil("class_name"),
-        "DeriveExport".to_variant()
+    // `class_name` should be empty for non-Object variants.
+    check_property(&property, "class_name", "");
+    check_property(&property, "type", VariantType::Int as i32);
+    check_property(&property, "hint", PropertyHint::PROPERTY_HINT_ENUM.ord());
+    check_property(&property, "hint_string", "A:0,B:1,C:2");
+    check_property(
+        &property,
+        "usage",
+        PropertyUsageFlags::PROPERTY_USAGE_DEFAULT.ord(),
     );
-    assert_eq!(
-        property.get_or_nil("type"),
-        (VariantType::Int as i32).to_variant()
+}
+
+#[derive(GodotClass)]
+#[class(init, base=Resource)]
+pub struct CustomResource {}
+
+#[godot_api]
+impl CustomResource {}
+
+#[godot_api]
+impl ResourceVirtual for CustomResource {}
```

Comment:
The whole `impl` is not needed anymore 😉 

(the inherent `impl` above is still required due to a bug, though...)

Reply:
Neat, when did that happen? Thought I was getting errors for not having that pretty recently

Reply:
The inherent one is still needed, but not the trait one.